spec/02-integration/01-helpers/05-rpc-mock_spec.lua

```diff
@@ -0,0 +1,154 @@
+local helpers = require("spec.helpers")
+local cp = require("spec.helpers.rpc_mock.cp")
+local dp = require("spec.helpers.rpc_mock.dp")
+local misc = require("spec.internal.misc")
+local get_node_id = misc.get_node_id
+
+local function trigger_change()
+  -- the initial sync is flaky. let's trigger a sync by creating a service
+  local admin_client = helpers.admin_client()
+  assert.res_status(201, admin_client:send {
+    method = "POST",
+    path = "/services/",
+    body = {
+      url = "http://example.com",
+    },
+    headers = {
+      ["Content-Type"] = "application/json",
+    },
+  })
+end
+
+describe("rpc mock/hook", function()
+  local recover
+  lazy_setup(function()
+    misc.repatch_timer()
+    if not kong.worker_events then
+      misc.patch_worker_events()
+      recover = true
+    end
+  end)
+
+  lazy_teardown(function()
+    misc.unrepatch_timer()
+    if recover then
+      kong.worker_events = nil
+    end
+  end)
+
+  describe("CP side mock", function()
+    local mocked_cp
+
+    lazy_setup(function()
+      helpers.get_db_utils()
+
+      mocked_cp = cp.new()
+      assert(mocked_cp:start())
+
+      assert(helpers.start_kong({
+        database = "off",
+        role = "data_plane",
+        cluster_mtls = "shared",
+        cluster_cert = "spec/fixtures/kong_clustering.crt",
+        cluster_cert_key = "spec/fixtures/kong_clustering.key",
+        nginx_conf = "spec/fixtures/custom_nginx.template",
+        cluster_rpc = "on",
+        cluster_rpc_sync = "on",
+        log_level = "debug",
+        cluster_control_plane = "127.0.0.1:8005",
+      }))
+    end)
+
+    lazy_teardown(function()
+      mocked_cp:stop(true)
+      helpers.stop_kong(nil, true)
+    end)
+
+    it("interception", function()
+      -- this will eventually cause DP to make a "kong.sync.v2.get_delta" call to the CP
```

Comment:
```suggestion
      -- POST /service entity to CP admin API
      -- this will eventually cause DP to make a "kong.sync.v2.get_delta" call to the CP
```